src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -23,47 +29,241 @@ def _graph_client_factory(**_):
     configure_common_settings(client)
     return client
 
+def list_role_definitions(name=None, resource_group_name=None, resource_id=None,
+                          custom_role_only=False):
+    definitions_client = _auth_client_factory().role_definitions
+    scope = _build_role_scope(resource_group_name, resource_id,
+                              definitions_client.config.subscription_id)
+    return _search_role_definitions(definitions_client, name, scope, custom_role_only)
+
+def create_role_definition(role_definition):
```

Comment:
After our discussion it seems like this can be achieved with multiple parameters. I would just add you might want to preserve a parameter to get the definition from the JSON file. 

---
In reply to: [73604232](https://github.com/Azure/azure-cli/pull/602#discussion_r73604232) [](ancestors = 73604232)